src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_IsPatternOperator.cs

```diff
@@ -17,20 +17,22 @@ public override BoundNode VisitIsPatternExpression(BoundIsPatternExpression node
             BoundDecisionDag decisionDag = node.GetDecisionDagForLowering(_factory.Compilation);
             bool negated = node.IsNegated;
             BoundExpression result;
-            if (canProduceLinearSequence(decisionDag.RootNode, whenTrueLabel: node.WhenTrueLabel, whenFalseLabel: node.WhenFalseLabel))
+            if (IsFailureNode(decisionDag.RootNode, node.WhenFalseLabel))
```

Comment:
Keeping the previous logic here but limiting the inverted scenario to immediate failures to not capture an `or` sequence.